doc/tor.1.txt

```diff
@@ -3044,6 +3044,32 @@ The next section describes the per service options that can only be set
    The HAProxy version 1 protocol is described in detail at
    https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
 
+[[HiddenServiceExportRendPoint]] **HiddenServiceExportRendPoint** **0**|**1**::
+   When HiddenServiceExportCircuitID along with this feature is enabled,
+   export IP and port of the rendezvous point along with the circuit ID. +
+ +
+   When this feature is enabled, tor will export IP and port of the rendezvous
+   point in the following way. +
+ +
+   "PROXY TCP6 fc00::AAAA:BBBB:CCCC:0:ffff:ffff ::1 65535 42\r\n" +
+ +
+      ip = (0xAAAA << 16) + 0xBBBB; +
+      port = 0xCCCC; +
+ +
+   For example, if the IP is 0x7f000001 and the port is 0x1bb, you can
+   translate to 127.30.0.1:443. (Default: 0)
```

Comment:
Am I missing something here or will this not work with IPv6?

Reply:
Good point. As Tor will move on to accept IPv6-only gradually, as agreed I think it's better to add first 12 chars of relay fingerprint instead of the (ip, orport) tuple.

Reply:
Fixed in `5399de4`.

Reply:
That's ok, but you should be aware that the fingerprint is SHA1(encoded(relay-public-key)). So it's relatively easy to brute-force 48 bits of that key.

So unlike an IPv4 address, it won't be possible to uniquely identify most relays using the partial fingerprint:
* there can be two relays in the consensus with the same 48 bit prefix
* there can be a relay in the consensus, and an attacker can set up an unpublished relay with the same fingerprint prefix

If that fits your use case, that's ok. But we do need to document that the partial fingerprint does not allow you to reliably identify a relay.

Reply:
@teor2345 How can there can be two relays in the consensus with the same fingerprint? Using the same relay key on different machines?

Reply:
The directory authorities only put one relay in the consensus for each unique fingerprint. But this code only uses 48 out of the 160 fingerprint bits. That's enough to identify most relays, but a determined attacker could fake those 48 fingerprint bits. (Also, SHA1 is being phased out, because it is insecure even when you use all 160 bits.)

So we need to document that it's usually right, but not cryptographically secure.

Reply:
I think it's probably easiest if I fix this documentation?

Reply:
Fixed in 8ef44ca.